convokit/model/corpusComponent.py

```diff
@@ -4,30 +4,79 @@
 
 
 class CorpusComponent:
-    def __init__(self, obj_type: str, owner=None, id=None, vectors: List[str] = None, meta=None):
+    def __init__(
+        self,
+        obj_type: str,
+        owner=None,
+        id=None,
+        initial_data=None,
+        vectors: List[str] = None,
+        meta=None,
+    ):
         self.obj_type = obj_type  # utterance, speaker, conversation
         self._owner = owner
+        self._id = id
+        self.vectors = vectors if vectors is not None else []
+
+        # if the CorpusComponent is initialized with an owner set up an entry
+        # in the owner's storage; if it is not initialized with an owner
+        # (i.e. it is a standalone object) set up a dict-based temp storage
+        if self.owner is None:
+            self._temp_storage = initial_data if initial_data is not None else {}
+        else:
+            self.owner.storage.initialize_data_for_component(
+                self.obj_type,
+                self._id,
+                initial_value=(initial_data if initial_data is not None else {}),
+            )
+
```

Comment:
At the moment, given how `__repr__` is written, this results in a representation that's confusing:

```
speaker = Speaker(id='Penelope')
utt = Utterance(id='333', speaker=speaker, text='falapooza')
utt
```

Utterance({'obj_type': 'utterance', 'vectors': [], 'meta': {}, 'speaker_': Speaker({'obj_type': 'speaker', 'vectors': [], 'meta': {}, 'owner': None, 'id': 'Penelope', 'temp_storage': {}}), 'owner': None, 'id': '333', 'temp_storage': {'speaker_id': 'Penelope', 'conversation_id': None, 'reply_to': None, 'timestamp': None, 'text': 'falapooza'}})

We should hide the implementation details, i.e. `speaker_` -> `speaker`, `temp_storage` should be de-structed so its variables are part of the highest-level dict.

Reply:
Likewise, for the other corpus component classes.

Reply:
I would lean towards opposing this change on the grounds that the purpose of `__repr__` is to provide a "real" representation (hence its name) of the object, that is useful for development and debugging - or, to quote one StackOverflow user, " `__repr__` is for developers, `__str__` is for customers" (See full answers in https://stackoverflow.com/questions/1436703/what-is-the-difference-between-str-and-repr). In this sense, hiding the presence of `temp_storage`, for example, runs counter to the purpose of `__repr__` and you can easily imagine such a decision getting in the way of someone trying to debug (that could be us!).

If you want a human-readable "pretty printed" output, that's the job of `__str__`, not `__repr__`. One thing I will concede is that `Conversation` and `Speaker` do not implement `__str__`, so that's certainly something we can fix (`Utterance` already implements `__str__`)

Reply:
Fair point, sounds good to me!